.github/workflows/ci.yaml

```
@@ -210,7 +210,32 @@ jobs:
           echo "${VERSION}"
           make build-${{ matrix.dist }}
 
-  ### e2e tests ###
+  build-gpu-operator-bundle-image:
+    if: github.ref == 'refs/heads/main' # Runs only if the event is from a main branch commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out code
+      - name: Calculate build vars
+        id: vars
+        run: |
+          echo "BRANCH_NAME=${GITHUB_REF_NAME}" >> $GITHUB_ENV
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build bundle-image
+        env:
+          BUNDLE_IMAGE: "ghcr.io/nvidia/gpu-operator/gpu-operator-bundle:${{ env.BRANCH_NAME }}-latest"
```

Why not this? If this works, then we don't need L219-222
```suggestion
          BUNDLE_IMAGE: "ghcr.io/nvidia/gpu-operator/gpu-operator-bundle:${{ github.ref_name }}-latest"
```

---

@tariq1890 
This approach is followed for future branch modifications.
If we consider both the release(feature) and main branches, we need to add one 'or' condition.
We don't need to change the code in the run

if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/release'}}

---

used GITHUB_REF_NAME